src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/AspNetWebApi2Integration.cs

```diff
@@ -198,7 +201,7 @@ public static class AspNetWebApi2Integration
             }
         }
 
-        private static Scope CreateScope(object controllerContext, out AspNetTags tags)
+        internal static Scope CreateScope(IHttpControllerContext controllerContext, out AspNetTags tags)
```

Comment:
Can this proxy DuckType be changed from an interface to a struct?

Reply:
In theory yes, but the object graph is quite deep: 

```
IHttpControllerContext
 - RouteData -> Route -> RouteTemplate
 - Request -> Method -> Method
```
Without data, I'm a bit afraid that the cost of the massive struct could outweigh the benefits. I'd rather go for the easiest solution for now, and we can come back to it and write some benchmarks when we have more time.